plonk-core/src/test.rs

```diff
@@ -18,15 +88,38 @@ macro_rules! batch_test {
                 #[test]
                 #[allow(non_snake_case)]
                 fn [< $test_set _on_ $engine>]() {
-                    $test_set::<$engine, $params>()
+                    $test_set::<<$engine as ark_ec::PairingEngine>::Fr, $params, crate::commitment::KZG10<$engine>>()
                 }
             )*
             $(
                 #[test]
                 #[should_panic]
                 #[allow(non_snake_case)]
                 fn [< $test_panic_set _on_ $engine>]() {
-                    $test_panic_set::<$engine, $params>()
+                    $test_panic_set::<<$engine as ark_ec::PairingEngine>::Fr, $params, crate::commitment::KZG10<$engine>>()
+                }
+            )*
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! batch_test_ipa {
+    ( [$($test_set:ident),*], [$($test_panic_set:ident),*] => ($curve:ty, $params:ty) ) => {
+        paste::item! {
+            $(
+                #[test]
+                #[allow(non_snake_case)]
+                fn [< $test_set _on_ $curve _ipa>]() {
+                    $test_set::<<$curve as ark_ec::PairingEngine>::Fr, $params, ark_poly_commit::ipa_pc::InnerProductArgPC<<$curve as ark_ec::PairingEngine>::G1Affine, blake2::Blake2s, ark_poly::univariate::DensePolynomial<<$curve as ark_ec::PairingEngine>::Fr>>>()
+                }
+            )*
+            $(
+                #[test]
+                #[should_panic]
+                #[allow(non_snake_case)]
+                fn [< $test_panic_set _on_ $curve _ipa>]() {
+                    $test_panic_set::<<$curve as ark_ec::PairingEngine>::Fr, $params, ark_poly_commit::ipa_pc::InnerProductArgPC<<$curve as ark_ec::PairingEngine>::G1Affine, blake2::Blake2s, ark_poly::univariate::DensePolynomial<<$curve as ark_ec::PairingEngine>::Fr>>>()
```

Comment:
Considering that we test with IPA the same exact number of test cases than with KZG, wouldn't be easier to include the call to `batch_test_ipa!` inside the `batch_test!` macro? 

We would reduce a lot of code duplicity.

Reply:
merged into `batch_test` then.

Reply:
Thanks!!!